discovery_engine_core/web-api/src/handlers.rs

```diff
@@ -14,33 +14,59 @@
 
 use std::convert::Infallible;
 use warp::hyper::StatusCode;
+use xayn_discovery_engine_ai::CoiSystemState;
+use xayn_discovery_engine_providers::Market;
 
 use crate::{
     db::Db,
-    models::{InteractionRequestBody, UserId},
+    models::{Article, InteractionRequestBody, UserId},
 };
 
-// TODO: TY-3013
-#[allow(clippy::unused_async)]
 pub(crate) async fn handle_ranked_documents(
-    _user_id: UserId,
-    _db: Db,
+    user_id: UserId,
+    db: Db,
 ) -> Result<impl warp::Reply, Infallible> {
-    Ok(StatusCode::NOT_IMPLEMENTED)
+    let default_state = CoiSystemState::default();
+    let user_interests = db.user_interests.read().await;
+    let state = user_interests.get(&user_id).unwrap_or(&default_state);
+
+    let mut documents = db.documents.clone();
+
+    db.coi.rank(&mut documents, &state.user_interests);
+
+    let articles = documents
+        .into_iter()
+        .map(|doc| Article::from(doc))
```

Comment:
```suggestion
        .map(Article::from)
```
This makes clippy fail but the CI is marked as passing